src/new-components/banner/banner.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react'
+import { Box } from '../box'
+import { Columns, Column } from '../columns'
+
+import styles from './banner.module.css'
+
+export type BannerTone = 'info' | 'promotion'
+
+type BannerProps = {
+    id?: string
+    /**
+     * The tone of the Banner. Affects the background color and the outline.
+     */
+    tone: BannerTone
+    /**
+     * The icon that should be added inside the Banner.
+     */
+    icon: React.ReactChild
+    /**
+     * The title to be displayed at the top of the Banner.
+     */
+    title: string
+    /**
+     * An optional description to be displayed inside the Banner.
+     */
+    description?: string
+    /**
+     * An optional action to displayed inside the Banner.
+     */
+    action?: React.ReactChild
+}
+
+function Banner({ id, tone, icon, title, description, action }: BannerProps) {
+    return (
+        <Box
+            id={id}
+            role="banner"
+            aria-live="polite"
+            borderRadius="standard"
+            className={[styles.banner, styles[`banner-${tone}`]]}
+        >
+            <Columns space="medium" alignY="center">
+                <Column width="content" aria-hidden>
+                    <Box display="flex" className={styles.icon} aria-hidden>
+                        {icon}
+                    </Box>
+                </Column>
+                <Column>
+                    <Box paddingY="xsmall" paddingRight={'small'}>
+                        {description ? (
+                            <Box className={[styles.title, styles[`title-${tone}`]]}>{title}</Box>
+                        ) : (
+                            <Box
+                                className={[
+                                    styles.title,
+                                    // If the banner does not have a description, we need to slightly tweak
+                                    // the styling of the title applying an extra css class
```

Comment:
Note to self: Errr there must be a better way to do this 😅

Reply:
If I understood correctly, this is about making the title bold only when a description is present, right? If so, check out how the `Toast` component does it:

https://github.com/Doist/reactist/blob/62102eb395c7e121904461d44646ff261c96a948/src/new-components/toast/static-toast.tsx#L96-L103

It uses the `Text` component, though. I'm not sure how that would affect what you're doing here. For instance, the `Text` component would need some work to make it adopt the text colors you need. I have some ideas on how to do that somewhat decently, but I have no hard objections to this approach. We can ship it and improve it later if needed. As long as it outwardly achieves what it's supposed to, that's ok.